core/domain/feedback_services.py

```diff
@@ -282,14 +291,36 @@ def get_all_threads(exploration_id, has_suggestion):
     return all_threads
 
 
-def enqueue_feedback_message_email_task(user_id):
-    """Adds a 'send feedback email' task into the taskqueue."""
+def get_all_thread_participants(exploration_id, thread_id):
+    """Returns set of all participants of a thread."""
+    return set([m.author_id for m in get_messages(exploration_id, thread_id)])
+
+
+def enqueue_feedback_message_batch_email_task(user_id):
+    """Adds a 'send feedback email' (batch) task into the taskqueue."""
 
     taskqueue_services.enqueue_task(
         feconf.FEEDBACK_MESSAGE_EMAIL_HANDLER_URL, {'user_id': user_id},
         feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS)
 
 
+def enqueue_feedback_message_instant_email_task(
+        user_id, reference, status_changed, old_status, new_status):
+    """Adds a 'send feedback email' (instant) task into the taskqueue."""
+
+    payload = {
+        'user_id': user_id,
+        'reference': reference
```

Comment:
Actually this dict is created from domain object, if you will see send_instant_emails() functions then you will see that initial reference is domain object. Then it converted to dict using to_dict() method and then saved in payload. I do not understand what do you mean by working with objects. Currently `reference` contains a dict inside it. Do you mean to replace that dict with an object ?